src/main/java/com/amazon/opendistroforelasticsearch/security/compliance/ComplianceConfig.java

```diff
@@ -162,38 +158,35 @@ public void log(Logger logger) {
-        final boolean logReadMetadataOnly = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_READ_METADATA_ONLY, false);
-        final boolean logWriteMetadataOnly = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_WRITE_METADATA_ONLY, false);
-        final boolean logDiffsForWrite = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_WRITE_LOG_DIFFS, false);
-        final List<String> watchedReadFields = settings.getAsList(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_READ_WATCHED_FIELDS,
-                Collections.emptyList(), false);
-        final List<String> watchedWriteIndices = settings.getAsList(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_WRITE_WATCHED_INDICES, Collections.emptyList());
+        final AuditConfig.Compliance compliance = AuditConfig.from(settings).getCompliance();
+        return from(compliance, settings);
+    }
+
+    /**
+     * Create compliance configuration from audit
+     * opendistrosecurityIndex - used to determine if internal index is written to or read from.
+     * type - checks if log destination used is internal elasticsearch.
+     * index - the index used for storing audit logs to avoid monitoring it.
+     * @param configCompliance configCompliance
+     * @param settings settings
+     * @return ComplianceConfig
+     */
+    public static ComplianceConfig from(AuditConfig.Compliance configCompliance, Settings settings) {
         final String opendistrosecurityIndex = settings.get(ConfigConstants.OPENDISTRO_SECURITY_CONFIG_INDEX_NAME, ConfigConstants.OPENDISTRO_SECURITY_DEFAULT_CONFIG_INDEX);
         final String type = settings.get(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_TYPE_DEFAULT, null);
         final String index = settings.get(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_CONFIG_DEFAULT_PREFIX + ConfigConstants.OPENDISTRO_SECURITY_AUDIT_ES_INDEX, "'security-auditlog-'YYYY.MM.dd");
-        final Set<String> ignoredComplianceUsersForRead = AuditConfig.getSettingAsSet(
-                settings,
-                ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_READ_IGNORE_USERS,
-                AuditConfig.DEFAULT_IGNORED_USERS,
-                false);
-        final Set<String> ignoredComplianceUsersForWrite = AuditConfig.getSettingAsSet(
-                settings,
-                ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_WRITE_IGNORE_USERS,
-                AuditConfig.DEFAULT_IGNORED_USERS,
-                false);
 
         return new ComplianceConfig(
-                logExternalConfig,
-                logInternalConfig,
-                logReadMetadataOnly,
-                watchedReadFields,
-                ignoredComplianceUsersForRead,
-                logWriteMetadataOnly,
-                logDiffsForWrite,
-                watchedWriteIndices,
-                ignoredComplianceUsersForWrite,
+                configCompliance.isComplianceEnabled(),
```

Comment:
This constructor should take `configCompliance` and store it as a member variable instead of copying each individual value.